templates/user/view/block/user-name.php

```diff
@@ -3,7 +3,11 @@
 defined( 'ABSPATH' ) || exit;
 
 $display = get_option( 'hp_user_enable_display' );
+?>
 
+<span class="hp-activity-badge <?php echo in_array( $user->get_id(), array_keys( (array) hivepress()->cache->get_cache( 'online_users' ) ) ) ? 'hp-activity-badge--online' : ''; ?>"></span>
```

Comment:
Please use request context instead, it's purpose is for values like this one, because cache still exposes the implementation. If we use request context then it's implementation-agnostic, just accessing the value, and if we leave the repeated cache fetching then we'd have to use some hooks or other hacky ways to alter the implementation because people will copy these template parts to child themes

Reply:
You can also add user ID to a list of IDs in the same callback that you'll add for the hivepress/v1/components/request/context hook, so there will be a single callback and also setting the values in the context

Reply:
Also (array) casting will not be needed because you'll ensure that the request context value is an array (even if empty) in the initial place where it's set